debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlValueConverters.java

```diff
@@ -789,4 +800,33 @@ public static Duration stringToDuration(String timeString) {
                     .minusNanos(nanoSeconds);
         }
     }
+
+    public static LocalDate stringToLocalDate(String dateString) {
+        final Matcher matcher = DATE_FIELD_PATTERN.matcher(dateString);
+        if (!matcher.matches()) {
+            throw new RuntimeException("Unexpected format for DATE column: " + dateString);
+        }
+
+        final int year = Integer.parseInt(matcher.group(1));
+        final int month = Integer.parseInt(matcher.group(2));
+        final int day = Integer.parseInt(matcher.group(3));
+
+        if (year == 0 || month == 0 || day == 0) {
+            return null;
+        }
+        return LocalDate.of(year, month, day);
+    }
+
+    public static boolean isValidTimestamp(String timestampString) {
```

Comment:
The notion of "valid" is a bit overloaded here as you also have the case where the value doesn't satisfy the regex. `containsZeroValues()` perhaps to make clearer what it is about?